lib/rubocop/cop/sorbet/obsolete_strict_memoization.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require "rubocop"
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # This cop checks the obsolete pattern for initializing instance variables that was required for older Sorbet
+      # versions in `#typed: strict` files.
+      #
+      # It's no longer required, as of Sorbet 0.5.10210.20220725105033-6e80c502e
+      # See https://github.com/sorbet/sorbet/issues/995
+      #
+      # @example
+      #
+      #   # bad
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo = T.let(@foo, T.nilable(Foo))
+      #     @foo ||= Foo.new
+      #   end
+      #
+      #   # good
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo ||= T.let(Foo.new, T.nilable(Foo))
+      #   end
+      #
+      # TODO: disable this cop when the Sorbet version is older than `0.5.10210.20220725105033-6e80c502e`.
```

Comment:
How do you do that? Is there something like a `support_target_sorbet_runtime_version?`

https://github.com/rubocop/rubocop/blob/172cc752c39a15439730adb5d8a75058ac18ecf4/lib/rubocop/cop/team.rb#L159-L180